Basic-Car-Maintenance/Shared/Dashboard/Views/EditEventDetailView.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  EditEventDetailView.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Aaron Wilson on 10/5/23.
+//
+
+import SwiftUI
+
+struct EditMaintenanceEventView: View {
+    @Binding var selectedEvent: MaintenanceEvent
+    @ObservedObject var viewModel: DashboardViewModel
+    @State private var title = ""
+    @State private var date = Date()
+    @State private var notes = ""
+    @Environment(\.dismiss) var dismiss
+    
+    var body: some View {
+        NavigationStack {
+            Form {
+                Section {
+                    TextField("Title",
+                              text: $title)
+                } header: {
+                    Text("Title")
+                }
+                
+                DatePicker(selection: $date, displayedComponents: .date) {
+                    Text("Date")
+                }
+                
+                Section {
+                    TextField("Notes", text: $notes, prompt: Text("Additional Notes"), axis: .vertical)
+                } header: {
+                    Text("Notes")
+                }
+            }
+            .onAppear(perform: {
+                setMaintenanceEventValues(event: selectedEvent)
+            })
+            .navigationTitle(Text("Add Maintenance"))
+            .toolbar {
+                ToolbarItem(placement: .topBarTrailing) {
+                    Button {
+                        dismiss()
+                    } label: {
+                        Text("Cancel")
+                    }
+                }
+                ToolbarItem(placement: .topBarLeading) {
+                    Button {
+                        var event = MaintenanceEvent(title: title, date: date, notes: notes)
+                        event.id = selectedEvent.id
+                        print("button pressed")
+                        Task {
+                            await viewModel.updateEvent(event)
+                        }
+                        dismiss()
```

Comment:
You don't need to create a new event object you should be able to pass `selectedEvent` directly, and given that you don't need 3 new state properties for `title`, `date`, and `notes`, you should be able to use `selectedEvent`

See below:
```swift
TextField("Title", text: $selectedEvent.title)
```

Reply:
And the `dismiss` should be inside of the `Task` so that the view doesn't dismiss if there was an error

Reply:
First of all, thank you so much for the detailed review! I couldn't watch live but I just rewatched the stream.

As for this issue, when I attempted to update the selectedEvent values directly, I ran into errors because the MaintenanceEvent properties are let constants. So rather than refactoring the model, I created a new object with the updated values and then in firebase overwrote the old object with the new one with the updated values.

Is there another approach I should use instead? 

I appreciate all your feedback.

Reply:
I'd say go ahead and change the model 👍